spec/routes_spec.rb

```diff
@@ -17,6 +19,41 @@
           it 'method defined' do
             expect(subject.respond_to?(action)).to be_truthy
           end
+
+          describe 'RBAC' do
+            before(:all) { MiqProductFeature.seed_features }
+
+            before do
+              allow(PrivilegeCheckerService).to receive(:new).and_return(check_service)
+              allow(check_service).to receive(:valid_session?).and_return(true)
+              allow(subject).to receive(:session)
+              allow(subject).to receive(:current_user)
+              allow(subject).to receive(:params).and_return({})
+            end
+
+            let(:check_service) { double(PrivilegeCheckerService) }
+
+            ignore.key?(controller.to_s) && ignore[controller.to_s].include?(action)
+
+            it 'is enforced' do
+              pending('ignored') if ignore.key?(controller.to_s) && ignore[controller.to_s].include?(action)
```

Comment:
I don't think we should mark it pending, because that implies we will add it sometime in the future.  If these are truly false positives, we should probably avoid creating the `it` at all (or even the entire `describe`)

Reply:
My concern is that people will keep private notes about how to add a new route where will be a checkmark for adding it to this list...

Reply:
Not sure I understand the concern...we can see if they add a false positive to the config during review and we would reject the PR.

Reply:
My concern is the human factor :rofl: also I think it's a good thing to see if a suspected false positive passes (rendering it to a regularly covered case) which we would not see if we never run it.

Reply:
As an example, we have similar concerns where people might add things to the brakeman config to mark things as false positives, and we just don't let them (in actuality, developers just know not to do that and instead try to actually fix the brakeman error they've created)

Reply:
> I think it's a good thing to see if a suspected false positive passes (rendering it to a regularly covered case) which we would not see if we never run it.

I'd be good with that, but then in that case I would let it be a regular passing spec as opposed to pending.  Pending just ends up flooding the pending section of the output and if it's never going to be removed from that section it's just noise.

Reply:
Well, this runs as a separate test case, so it's not flooding the main one :thinking:

Reply:
That is, I'd change this test to something like the following:

```ruby
was_expected =
  expect do
    subject.send(:check_privileges)
    subject.send(action)
  rescue MiqException::RbacPrivilegeException
    raise
  rescue => ex
    # NOP: we don't care if any other exception happens
  end

if ignore[controller.to_s]&.include?(action)
  was_expected.to_not raise_error(MiqException::RbacPrivilegeException)
else
  was_expected.to raise_error(MiqException::RbacPrivilegeException)
end
```

Reply:
Yeah, that doesn't mark the false positives that nicely :confused:

Reply:
I'm not sure what you are trying to solve/not solve... false positive are clearly outlined in the ignore file... can we discuss?

Reply:
Ok discussed with @skateman and it turns out that these are not being ignored as false positives, but more like a TODO list since they are more complicated to test.  As such we agreed to rename the `ignore` terminology to `pending` both in the file name and the variable name.  If we later have actual false positives we can have an actual ignore file.